src/Database/Bloodhound/Types.hs

```diff
@@ -1744,9 +1746,20 @@ class BucketAggregation a where
 
 data Bucket a = Bucket { buckets :: [a]} deriving (Show)
 
-data TermsResult = TermsResult { termKey       :: Text
-                               , termsDocCount :: Int
-                               , termsAggs     :: Maybe AggregationResults } deriving (Show)
+data TermsResult
+  = TextTermsResult
```

Comment:
Ah yes, makes sense. Addressed in https://github.com/MHova/bloodhound/commit/b9924936228a8826806b53eb3927f249f014ff00